client/src/Routes.js

```diff
@@ -1,21 +1,32 @@
 import React from "react";
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
-
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import { useAuth } from "./hooks/useAuth";
 import FormContainer from "./components/Form/FormContainer";
+import MessagerContainer from "./components/Messager/MessagerContainer";
+import PrivateRoute from "./components/PrivateRoute";
 
 export default function Routes() {
+  const auth = useAuth();
```

Comment:
I would recommend adding a loading state to the auth hook, and showing a loading spinner while the user is being authorized. It would prevent the login page from popping up for a second before the messenger page appears, if the user is logged in.

Reply:
Great suggestion! That flashing login was a little annoying and this is a great fix for that! Will definitely help too if I ever deployed the server! Done!